Sources/Networking/SessionManager/SessionManager.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  SessionManager.swift
+//
+//  Created by Filip Gulan on 26/02/2020.
+//  Copyright © 2020 Infinum. All rights reserved.
+//
+
+import Foundation
+import Alamofire
+
+/// Use it to store all kind of session related info like authorization adapter, networking session,
+/// currently logged in user, etc.
+public class SessionManager {
+
+    // MARK: - Public properties
+
+    /// Shared networking manager
+    public static let `default` = SessionManager()
+    
+    /// Authorization adapter will be prefixed to all requests
+    public var authorizationAdapter: RequestInterceptor?
+    
+    /// Alamofire Session storage
+    public var session: Session { _session }
+    
+    // MARK: - Private storage
+    
+    // Avoid possibility of someone changing session outside of manager scope.
+    // We have to use `lazy var` because of configuration extraction
+    private lazy var _session: Session = {
+        return Session(
+            configuration: defaultConfiguration,
+            interceptor: interceptor,
+            serverTrustManager: trustManager,
+            eventMonitors: eventMonitors
+        )
+    }()
+    
+    private init() { }
+}
+
+// MARK: - Private inits
+
+private extension SessionManager {
+    
+    // Set up custom session configuration, for example: Loggie usage.
+    var defaultConfiguration: URLSessionConfiguration {
+        let configuration: URLSessionConfiguration = .default
```

Comment:
Why not use alamofire default session configuration?
```suggestion
        let configuration: URLSessionConfiguration = .af.default
```

Reply:
Yeah I could but I used this way since most people will add Loggie or something similar to their project and with this approach it is a single line change (for example, you just pass .loggie configuration which doesn't have default headers set yet)

```swift
let configuration: URLSessionConfiguration = useLoggie ? .loggie : .default
```